paddle/fluid/framework/details/op_handle_base.cc

```diff
@@ -78,6 +110,39 @@ void OpHandleBase::RecordWaitEventOnCtx(platform::DeviceContext *waited_ctx) {
 #endif
 }
 
+void OpHandleBase::RecordWaitEventOnCtx2(
```

Comment:
Why need to add this? Note that experiments show that in distributed training with nccl2, wait allreduce to finish slows down the total throughput.